### PR TITLE
GameObject -> BloonScript reference

### PR DESCRIPTION
**Pull request by SSeelos**
you can reference monobehaviours instead of gameobjects so you dont have to call get component (wich might not even be on it)

### CODE REVIEW

**Comment by Serpendio** [comment]
What if for this one I know that I wish to also instantiate with the Rigidbody & SpriteRenderer? 
Sure, I could AddComponent, but using a prefab is both simpler to quickly modify and less memory intensive. No?

**Comment by SSeelos** [comment]
that should not be a problem. the reference will still point to the prefab where the monobehaviour is added to. so you can just call get component on the component itself just as you would with a gameobject reference
image

File: Assets/Scripts/BloonSpawner.cs [hunk 1]
```diff
@@ -48,8 +48,8 @@ public void SpawnWave()
     private IEnumerator SpawnBloon(float spawnDelay)
     {
         yield return new WaitForSeconds(spawnDelay);
-        Instantiate(bloonBase);
-        var script = bloonBase.GetComponent<BloonScript>();
+
+        var script = Instantiate(bloonBase);
```
Comment (SSeelos) [inline Assets/Scripts/BloonSpawner.cs:52]: my review

**Review by SSeelos — COMMENTED** [review]
my comment

File: Assets/ScriptTemplates/81-C# Script__MonoBehaviour-MB.cs.txt [hunk 2]
```diff
@@ -1,13 +1,8 @@
-using System.Collections;
-using System.Collections.Generic;
 using UnityEngine;
 
     #ROOTNAMESPACEBEGIN#
 public class #SCRIPTNAME# : MonoBehaviour
 {
-    void Awake()
```
Comment (SSeelos) [inline Assets/ScriptTemplates/81-C# Script__MonoBehaviour-MB.cs.txt]: i like having awake in there but i see your point
i use awake to set up dependencies (getcomponent) calls
then in start i might get properties of the cached components

File: Assets/ScriptTemplates/81-C# Script__ScriptableObject-SO.cs.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+using System.Collections;
+using System.Collections.Generic;
```
Comment (Serpendio) [inline Assets/ScriptTemplates/81-C# Script__ScriptableObject-SO.cs.txt]:
```suggestion
```
No need for these in most ScriptableObjects either

File: Assets/Scripts/BloonSpawner.cs (hunk 1 above)
Comment (SSeelos) [reply Assets/Scripts/BloonSpawner.cs:52]: my reply

**Review by Serpendio — CHANGES_REQUESTED** [review]

**Review by Serpendio — APPROVED** [review]

**Pull request merged**